### PR TITLE
[Issue #129] Fix duplicated META-INF/*.kotlin_module  issue

### PR DESCRIPTION
**Pull request by slmlt**

### CODE REVIEW

**Review by ZacSweers — COMMENTED** [review]
Let's DRY this up a bit by moving the KotlinCompile configuration to the root build file and have it run on subprojects, then make the module name reuse the artifact ID gradle property that each project already has (see their `gradle.properties`)

**Comment by slmlt** [comment]
Done :) Used POM_ARTIFACT_ID from gradle.properties

File: build.gradle.kts [hunk 1]
```diff
@@ -162,4 +162,12 @@ subprojects {
       }
     }
   }
+  // configuration required to produce unique META-INF/*.kotlin_module file name
```
Comment (ZacSweers) [inline build.gradle.kts:165]:
```suggestion
  // configuration required to produce unique META-INF/*.kotlin_module file names
```

**Review by ZacSweers — APPROVED** [review]

**Comment by ZacSweers** [comment]
Thanks! I plan to cut a release after beta05 is out, hopefully this week

**Comment by ZacSweers** [comment]
@slmlt when I try this locally, I see the following warnings. Are you familiar with them and what possible recourse there is?

```
> Task :moshi-sealed:codegen:kaptGenerateStubsKotlin
w: Argument -module-name is passed multiple times. Only the last value will be used: moshi-sealed-codegen

> Task :moshi-sealed:codegen-ksp:kspKotlin
w: Argument -module-name is passed multiple times. Only the last value will be used: moshi-sealed-ksp

> Task :moshi-sealed:codegen-ksp:compileKotlin
w: Argument -module-name is passed multiple times. Only the last value will be used: moshi-sealed-ksp

```

**Comment by slmlt** [comment]
@ZacSweers I assume, that KotlinCompile task has its own default arguments (`KotlinCompile.createCompilerArgs()`), including the "-module-name" (seems to come from `K2JVMCompilerArguments.moduleName`) which we cannot modify. By specifying additional compiler options (through freeCompilerArgs property) we rely on the fact, that for compiler options with matching names only the last value will be used (the very same statement we got warnings about), so the warnings are somewhat expected. Unfortunately this is the only workaround I was able to find.

**Comment by ZacSweers** [comment]
Do you know if there's a feature request on youtrack? Otherwise we should file one, this prevents use of -Werror

**Comment by slmlt** [comment]
Well, in a matter of fact there is an issue about this https://youtrack.jetbrains.com/issue/KT-42058 and it is already fixed. Now we actually can do this:
`
  kotlinOptions.moduleName = project.property("POM_ARTIFACT_ID") as String
`
instead of using freeCompilerArgs. Sorry, I obviously did not put enough effort to research the problem. Another PR https://github.com/ZacSweers/MoshiX/pull/134

**Pull request merged**